src/provider/provider_tracking.c

```diff
@@ -119,6 +120,134 @@ static umf_result_t trackingAlloc(void *hProvider, size_t size,
     return ret;
 }
 
+static umf_result_t trackingAllocationSplit(void *provider, void *ptr,
+                                            size_t totalSize, size_t leftSize) {
+    umf_tracking_memory_provider_t *p =
+        (umf_tracking_memory_provider_t *)provider;
+
+    tracker_value_t *splitValue =
```

Comment:
if you allocate `splitValue` just before `critnib_insert` then you do not need to `free(splitValue)` in every error cases below.